# 안녕하세요^^ 
## AIVLE 1차 미니프로젝트 '서울시 생활정보 기반 대중교통 수요 분석 ' 과정에 오신 여러분을 환영합니다.
* 본 과정에서는 실제 사례와 데이터를 기반으로 문제를 해결하는 전체 과정을 자기 주도형 실습으로 진행해볼 예정입니다.
* 앞선 교육과정을 정리하는 마음과 지금까지 배운 내용을 바탕으로 문제 해결을 해볼게요!
* 미니 프로젝트를 통한 문제 해결 과정 'A에서 Z까지', 지금부터 시작합니다!

---

# 데이터 분석부터 먼저 시작해보겠습니다.
## "구 별 유동인구 데이터" 를 확인해 보도록 하겠습니다

In [ ]:
# 필요 라이브러리부터 설치합니다.
# %pip install pandas seaborn

## 기본전제
* 처음에 제공되는 기본/추가 데이터는 '에이블러용' 폴더에 있습니다.

[기본 데이터]
* seoul_moving_month_202401.csv

[데이터 소개]
* 서울 시 구별 이동 2024년 1월 데이터
* https://data.seoul.go.kr/dataVisual/seoul/seoulLivingMigration.do

[변수 소개]
* 도착시간 : 0 -> "0:00~0:59"를 의미 
* 출발 자치구 : 자치구 코드
* 성별 : F(여성), M(남성)
* 나이 : 10 ~ 79세 (5세 단위), 0 ~ 10세, 80세 이상으로 구분
* 유형 : H(야간상주지), W(주간상주지), E(기타)
* 평균 이동 시간 : 동일 열 내 이동인구의 월 단위 평균 이동 시간 (10분단위)
* 이동인구 : 추정 합산값으로 소수점 표출, 개인정보 비식별화를 위해 3명 미만의 경우 * 처리

---

# 1.데이터 불러오기
## 모든 미니 프로젝트의 시작은 '데이터 불러오기' 부터라고 할 수 있습니다.
+ KeyPoint : 불러오고자 하는 데이터에 따라 자유롭게 변수로 지정할 수 있다.

###  데이터 프레임을 불러오고 변수로 저장(여기서는 CSV 기준으로 진행)
* csv : pd.read_csv("파일이름. csv")
* txt : pd.read_csv("파일이름. csv", sep="구분자")
* xlsx : pd.read_excel('파일이름.xlsx')
* pickle : pd.read_pickle("파일이름.pkl") <br>
 [참고] pickle은 파이썬의 모든 객체를 파일로 저장할 수 있는 방법으로 DataFrame,List,Dict 등 모든 객체 저장 가능(특히 sklearn라이브러리를 통해 모델을 학습시키고, 저장할 때 많이 사용)

#### [실습문제1] 데이터 로딩
* Pandas 라이브러리를 활용해서 'seoul_moving_month_202401.csv' 데이터를 확인하세요.
    * 아래에 'Pandas'를 Import하고 'seoul_moving_month_202401.csv'파일을 'seoul_moving'에 불러오고,  Shape을 확인하는 소스코드를 작성하고 실행해보세요.
    * 데이터 파일 로딩시 참고 사항 
        * 구분자(sep)는 ',' 입니다
        * UTF-8 인코더를 사용해 주세요

In [2]:
import os
os.getcwd()

'C:\\Users\\User\\KT_AIVLE\\프로젝트\\미니 프로젝트 1차'

In [30]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.

import pandas as pd
seoul_moving = pd.read_csv('./1.2 seoul_moving_month_202401.csv', encoding='UTF-8')
seoul_moving.head()

대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분) 이동인구(합)
0  202401  일     0      11010      11010  F   0   EE           20   28.06
1  202401  일     0      11010      11010  F  10   EE          167   11.96
2  202401  일     0      11010      11010  F  10   EH          203   17.63
3  202401  일     0      11010      11010  F  10   HE           23    5.96
4  202401  일     0      11010      11010  F  10   WH           27    5.90

In [28]:
seoul_moving.loc[:,'이동유형'].value_counts()

이동유형
EH    1814987
HE    1691265
EE    1660940
WH    1348812
HW    1186337
WE    1092337
EW     982178
WW     606279
HH     408634
Name: count, dtype: int64

In [143]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 데이터 프레임의 Shape을 확인합니다.
seoul_moving.shape

(29361031, 10)

---

# 2.기본 정보 확인 및 클렌징

+ 데이터 클렌징 : 결측치, 이상치 등을 제거하여 데이터 분석 결과가 왜곡 되는 문제를 방지하기 위한 정제 과정

#### [실습문제2] 기본 정보 확인하기
* 'seoul_moving' 데이터의 정보를 확인해보세요.
* 'describe', 'info', 'head' 등 전부 활용해 보겠습니다.

In [15]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# describe()
seoul_moving.describe()


대상연월      도착시간      출발 시군구 코드      도착 시군구 코드             나이  \
count  773834.0  773834.0  773834.000000  773834.000000  773834.000000   
mean   202401.0       0.0   15791.565226   17725.415084      38.315395   
std         0.0       0.0    8365.233399    9229.498952      16.236471   
min    202401.0       0.0   11010.000000   11010.000000       0.000000   
25%    202401.0       0.0   11090.000000   11110.000000      25.000000   
50%    202401.0       0.0   11190.000000   11210.000000      35.000000   
75%    202401.0       0.0   11250.000000   31023.000000      50.000000   
max    202401.0       0.0   39000.000000   39000.000000      80.000000   

         평균 이동 시간(분)  
count  773834.000000  
mean      101.290477  
std        88.299001  
min         1.000000  
25%        47.000000  
50%        76.000000  
75%       123.000000  
max      1581.000000

In [145]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# info()
seoul_moving.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29361031 entries, 0 to 29361030
Data columns (total 10 columns):
 #   Column       Dtype 
---  ------       ----- 
 0   대상연월         int64 
 1   요일           object
 2   도착시간         int64 
 3   출발 시군구 코드    int64 
 4   도착 시군구 코드    int64 
 5   성별           object
 6   나이           int64 
 7   이동유형         object
 8   평균 이동 시간(분)  int64 
 9   이동인구(합)      object
dtypes: int64(6), object(4)
memory usage: 2.2+ GB


In [156]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# head()
seoul_moving.rename(columns = {
    '출발 시군구 코드':'출발도시',
    '도착 시군구 코드':'도착도시',
    '평균 이동 시간(분)':'평균이동시간',
    '이동인구(합)':'이동인구'
},inplace=True)
seoul_moving.head()

대상연월 요일  도착시간   출발도시   도착도시 성별  나이 이동유형  평균이동시간   이동인구
0  202401  일     0  11010  11010  F   0   EE      20  28.06
1  202401  일     0  11010  11010  F  10   EE     167  11.96
2  202401  일     0  11010  11010  F  10   EH     203  17.63
3  202401  일     0  11010  11010  F  10   HE      23   5.96
4  202401  일     0  11010  11010  F  10   WH      27   5.90

* 1번 열의 의미
    * " 2024년 1월 일요일에 11010(종로구)에서 11010(종로구)으로 이동하여 0시 ~ 0시 59분 사이에 도착했고, E(기타)에서 E(기타)로 이동한 10 ~ 15세 여성은 총 11.96명이며 이들의 평균 이동시간은 167분이다


#### [실습문제3] 결측치 처리 ( NaN이 아닌 결측치를 처리 해봅시다)

* '이동인구(합)' column은 이동 인구의 합산인데 object 타입입니다. 

    - 왜 일까요? 

    - Tip! (이동인구 : 추정 합산값으로 소수점 표출, 개인정보 비식별화를 위해 3명 미만의 경우 * 처리)
    - 3명 미만의 이동인구는 * 처리 되어 있어서 해당 값을 나름의 기준으로 변경하세요.(중위값)
    - '이동인구(합)' 컬럼의 데이터 타입을 수치형으로 변경하세요 

In [17]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# value_counts()
seoul_moving.value_counts()

대상연월    요일  도착시간  출발 시군구 코드  도착 시군구 코드  성별  나이  이동유형  평균 이동 시간(분)  이동인구(합)
202401  금   0     11010      11010      F   10  EE    139          12.03      1
        일   0     25000      11240      M   40  EE    8            *          1
                                        F   60  EH    125          3.79       1
                                            65  EH    207          4.68       1
                                        M   10  EH    110          6.61       1
                                                                             ..
        수   0     11170      31050      F   35  HE    9            3.03       1
                                                HH    54           3.03       1
                                                WE    39           3.04       1
                                                WH    84           36.42      1
        화   0     39000      11250      M   70  EH    149          4.60       1
Name: count, Length: 773834, dtype: int64

In [31]:
# '*' 값의 최빈값과 평균값을 알 수 없는 상황이라 중위값으로 대체 (1.5명)
seoul_moving.loc[seoul_moving['이동인구(합)'] =='*', '이동인구(합)'] = '1.5'


In [32]:
# 이동인구(합) column의 데이터 타입을 바꿔주세요
# astype()
seoul_moving['이동인구(합)'] = seoul_moving['이동인구(합)'].astype(float)

#### [실습문제4] 이동유형, 요일, 성별 변수의 value수를 파악

* 역시나 value_counts()를 활용해 봅시다.

In [147]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 이동유형, value_counts()
seoul_moving['이동유형'].value_counts()

이동유형
EH    5947038
HE    5585412
EE    5224553
WH    3078348
WE    2759477
HW    2700535
EW    2330931
WW    1082641
HH     652096
Name: count, dtype: int64

* 기타 -> 집, 집 -> 기타, 기타 -> 기타 순으로 많다

In [148]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 요일, value_counts()
seoul_moving['요일'].value_counts()

요일
월    4448237
수    4341429
화    4300070
토    4254525
금    4251430
목    4005959
일    3759381
Name: count, dtype: int64

* 주말이 가장 인구유동이 많지만, 또이또이하다

* 24년 1월의 요일 수를 알려드릴게요
    * 일 : 4, 월 : 5, 화 : 5, 수 : 5, 목 : 4, 금 : 4, 토 : 4

In [149]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 성별, value_counts()
print(seoul_moving['성별'].value_counts())
seoul_moving['성별'].value_counts(normalize=True)

성별
M    16433525
F    12927506
Name: count, dtype: int64


성별
M    0.559705
F    0.440295
Name: proportion, dtype: float64

* 남성이 여성보다 더 많다.

#### [실습문제5] 어느 요일의 심야 이동이 가장 많을까요?

In [29]:
condition = !(temp['도착시간']<=23) & (temp['도착시간']>=5)
seoul_moving.loc[condition,:]

KeyError: "None of [Index(['지정된 파일을 찾을 수 없습니다.',\n       ''temp['도착시간']'은(는) 내부 또는 외부 명령, 실행할 수 있는 프로그램, 또는', '배치 파일이 아닙니다.'],\n      dtype='object')] are in the [index]"

In [152]:
seoul_moving.isna().sum()
# seoul_moving = pd.read_csv('../../데이터/seoul_moving.csv')

대상연월           0
요일             0
도착시간           0
출발 시군구 코드      0
도착 시군구 코드      0
성별             0
나이             0
이동유형           0
평균 이동 시간(분)    0
이동인구(합)        0
dtype: int64

In [157]:
seoul_moving.groupby('요일', as_index=False)['이동인구'].sum()

요일          이동인구
0  금  1.047175e+08
1  목  1.005787e+08
2  수  1.224716e+08
3  월  1.129803e+08
4  일  7.462348e+07
5  토  9.098288e+07
6  화  1.194038e+08

#### [실습문제6] 서울 지역의 데이터만 골라 보도록 하겠습니다.

* 우리는 서울시의 데이터만 필요합니다.
* 시군구 코드:광역시도(2자리)+시군구(3자리) / 서울:11, 부산:26, 대구:27, 인천:28, 광주:29, 대전:30, 울산:31)

* 출발과 도착이 모두 서울인 데이터만 추출해 주세요.

In [12]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 출발 시군구 코드, 도착 시군구 코드 <20000
condition = (seoul_moving['출발 시군구 코드'] < 20000)& (seoul_moving['도착 시군구 코드'] < 20000)
seoul_moving=seoul_moving.loc[condition, :]
seoul_moving.tail()
# seoul_moving.info()

대상연월 요일  도착시간  출발 시군구 코드  도착 시군구 코드 성별  나이 이동유형  평균 이동 시간(분)  \
29320192  202401  토    23      11250      11250  M  80   EH           18   
29320193  202401  토    23      11250      11250  M  80   HE           78   
29320194  202401  토    23      11250      11250  M  80   HW          168   
29320195  202401  토    23      11250      11250  M  80   WH           16   
29320196  202401  토    23      11250      11250  M  80   WW           78   

          이동인구(합)  
29320192    28.41  
29320193     9.46  
29320194     9.48  
29320195    47.27  
29320196    28.51

In [13]:
seoul_moving.rename(columns={'출발 시군구 코드':'출발자치구코드','도착 시군구 코드':'도착자치구코드'}, inplace = True)

C:\Users\User\AppData\Local\Temp\ipykernel_17224\423258872.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  seoul_moving.rename(columns={'출발 시군구 코드':'출발자치구코드','도착 시군구 코드':'도착자치구코드'}, inplace = True)


In [14]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
seoul_moving['출발자치구'] = seoul_moving['출발자치구코드'].map({
    11010 :'종로구',
11020 :'중구',
11030 :'용산구',
11040 :'성동구',
11050 :'광진구',
11060 :'동대문구',
11070 :'중랑구',
11080 :'성북구',
11090 :'강북구',
11100 :'도봉구',
11110 :'노원구',
11120 :'은평구',
11130 :'서대문구',
11140 :'마포구',
11150 :'양천구',
11160 :'강서구',
11170 :'구로구',
11180 :'금천구',
11190 :'영등포구',
11200 :'동작구',
11210 :'관악구',
11220 :'서초구',
11230 :'강남구',
11240 :'송파구',
11250 :'강동구',
})
seoul_moving['도착자치구'] = seoul_moving['도착자치구코드'].map({
    11010 :'종로구',
11020 :'중구',
11030 :'용산구',
11040 :'성동구',
11050 :'광진구',
11060 :'동대문구',
11070 :'중랑구',
11080 :'성북구',
11090 :'강북구',
11100 :'도봉구',
11110 :'노원구',
11120 :'은평구',
11130 :'서대문구',
11140 :'마포구',
11150 :'양천구',
11160 :'강서구',
11170 :'구로구',
11180 :'금천구',
11190 :'영등포구',
11200 :'동작구',
11210 :'관악구',
11220 :'서초구',
11230 :'강남구',
11240 :'송파구',
11250 :'강동구',
})
seoul_moving.tail()

C:\Users\User\AppData\Local\Temp\ipykernel_17224\1304125206.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  seoul_moving['출발자치구'] = seoul_moving['출발자치구코드'].map({
C:\Users\User\AppData\Local\Temp\ipykernel_17224\1304125206.py:29: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  seoul_moving['도착자치구'] = seoul_moving['도착자치구코드'].map({


대상연월 요일  도착시간  출발자치구코드  도착자치구코드 성별  나이 이동유형  평균 이동 시간(분)  이동인구(합)  \
29320192  202401  토    23    11250    11250  M  80   EH           18    28.41   
29320193  202401  토    23    11250    11250  M  80   HE           78     9.46   
29320194  202401  토    23    11250    11250  M  80   HW          168     9.48   
29320195  202401  토    23    11250    11250  M  80   WH           16    47.27   
29320196  202401  토    23    11250    11250  M  80   WW           78    28.51   

         출발자치구 도착자치구  
29320192   강동구   강동구  
29320193   강동구   강동구  
29320194   강동구   강동구  
29320195   강동구   강동구  
29320196   강동구   강동구

#### [실습문제7] 서울지역의 구 별로 groupby하여 데이터를 확인하겠습니다 (유동인구 기준)

* 유출이 제일 많은 구는?
강남구 117만

* 유입이 제일 많은 구는?
강남 55.7만
* 유출이 제일 적은 구는?
금천구 13.7만
* 유입이 제일 적은 구는?
금천구 17.8만

* 참고자료 서울 구별 코드
    * 11010	종로구
    * 11020	중구
    * 11030	용산구
    * 11040	성동구
    * 11050	광진구
    * 11060	동대문구
    * 11070	중랑구
    * 11080	성북구
    * 11090	강북구
    * 11100	도봉구
    * 11110	노원구
    * 11120	은평구
    * 11130	서대문구
    * 11140	마포구
    * 11150	양천구
    * 11160	강서구
    * 11170	구로구
    * 11180	금천구
    * 11190	영등포구
    * 11200	동작구
    * 11210	관악구
    * 11220	서초구
    * 11230	강남구
    * 11240	송파구
    * 11250	강동구

In [15]:
seoul_moving.shape

(10791769, 12)

In [17]:
seoul_moving.tail(3)

대상연월 요일  도착시간  출발자치구코드  도착자치구코드 성별  나이 이동유형  평균 이동 시간(분)  이동인구(합)  \
29320194  202401  토    23    11250    11250  M  80   HW          168     9.48   
29320195  202401  토    23    11250    11250  M  80   WH           16    47.27   
29320196  202401  토    23    11250    11250  M  80   WW           78    28.51   

         출발자치구 도착자치구  
29320194   강동구   강동구  
29320195   강동구   강동구  
29320196   강동구   강동구

In [18]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 유출이 제일 많은 구와 적은 구는?
temp = seoul_moving.loc[seoul_moving['출발자치구코드']!=seoul_moving['도착자치구코드'],:] 
condition = (temp['도착시간']<=23) & (temp['도착시간']>=5)
temp = temp.loc[condition, :]
moving_out_num = temp.groupby('출발자치구코드')['이동인구(합)'].sum()
moving_in_num = temp.groupby('도착자치구코드')['이동인구(합)'].sum()
moving = pd.concat([moving_out_num,moving_in_num], axis=1)


In [21]:

import numpy as np
moving.columns = ['유출인구','유입인구']
moving['전체유동인구'] = seoul_moving['유입인구'] - seoul_moving['유출인구'] 
moving.head()

KeyError: '유입인구'

In [209]:
moving = moving.reset_index()
moving.rename(columns={'index':'자치구'}, inplace=True)

In [164]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 유입이 제일 많은 구와 적은 구는?
print(moving['유입인구'].sort_values().head())
print('='*20)
print(moving['유입인구'].sort_values(ascending=False).head())
print('='*20)
print(moving['유출인구'].sort_values().head())
print('='*20)
print(moving['유출인구'].sort_values(ascending=False).head())

11180    4379891.58
11100    5684682.99
11070    6466159.98
11090    6474507.03
11120    6630722.83
Name: 유입인구, dtype: float64
11230    23271626.11
11220    16603565.69
11020    14078685.33
11190    13924518.07
11240    13260464.70
Name: 유입인구, dtype: float64
11180    4385750.01
11100    5686234.59
11070    6449369.23
11090    6482328.38
11120    6639329.34
Name: 유출인구, dtype: float64
11230    23297651.03
11220    16529995.81
11020    14049403.00
11190    13959033.76
11240    13349555.90
Name: 유출인구, dtype: float64


* 유출인구 1등 강남구
* 유입인구 1등 강남구
* 유출 꼴등 금천구
* 유입 꼴등 금천구

#### [실습문제8] 낮시간만 볼게요 심야버스 제외하겠습니다

* 도착시간 0부터 4까지(0:00 ~ 4:59) 제거

In [25]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# '도착시간' >=5
condition = (seoul_moving['도착시간']<=23) & (seoul_moving['도착시간']>=5)
seoul_day = seoul_moving.loc[condition, :]
seoul_day[-200:-100]

대상연월 요일  도착시간  출발자치구코드  도착자치구코드 성별  나이 이동유형  평균 이동 시간(분)  이동인구(합)  \
29319997  202401  토    23    11250    11250  F  45   EW           47    56.35   
29319998  202401  토    23    11250    11250  F  45   HE           41    88.85   
29319999  202401  토    23    11250    11250  F  45   HH           32   181.96   
29320000  202401  토    23    11250    11250  F  45   HW           48    53.02   
29320001  202401  토    23    11250    11250  F  45   WE           23    33.28   
...          ... ..   ...      ...      ... ..  ..  ...          ...      ...   
29320092  202401  토    23    11250    11250  M  20   HW           31    45.76   
29320093  202401  토    23    11250    11250  M  20   WE           41    59.33   
29320094  202401  토    23    11250    11250  M  20   WH           34   347.06   
29320095  202401  토    23    11250    11250  M  20   WW           55    43.06   
29320096  202401  토    23    11250    11250  M  25   EE           35   551.70   

         출발자치구 도착자치구  
29319997   강동구   강동구  
29319998   강동구   강동구  
29319999   강동구   강동구  
29320000   강동구   강동구  
29320001   강동구   강동구  
...        ...   ...  
29320092   강동구   강동구  
29320093   강동구   강동구  
29320094   강동구   강동구  
29320095   강동구   강동구  
29320096   강동구   강동구  

[100 rows x 12 columns]

[실습문제9] '도착 시군구 코드' 를 기준으로 각 구별로 '평균 이동 시간(분)' , '이동인구(합)' 의 평균 과 합을 구해보겠습니다

* groupby 사용, as_index=False로 설정해주세요

In [23]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 평균을 구해서 df_mean 변수로 저장
seoul_day.to_csv('./seoul_day.csv')
import pandas as pd
# seoul_day = pd.read_csv('./seoul_day.csv')
# 이동 인구의 평균은 ? 시간 단위로 계산하여 평균 내리기
seoul_day.groupby('도착자치구코드', as_index=False)['이동인구'].sum() #24-5 = 19 시간으로 나누면 시간당 평균 이동인구

KeyError: 'Column not found: 이동인구'

* 이동인구 평균을 구하기 위한 기준 단위가 무엇인가?

In [178]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.
# 합을 구해서 df_sum 변수로 저장
# 평균이동시간(분) -> 총 이동 시간, 이동인구(합) -> 총 이동인구로 이름 대체
df_sum = seoul_day.groupby('도착자치구코드', as_index=False)[['평균이동시간','이동인구']].sum()
df_sum.columns = ['도착자치구코드','이동시간합계','이동인구합계']

In [6]:
# 시간단위 이동인구 평균 및 이동시간 평균
# grouby 두 개
group_data = seoul_day.groupby('도착자치구코드', as_index=False)
group_data['이동인구'].describe()

도착자치구코드     count       mean         std  min   25%    50%    75%  \
0     11010  443076.0  44.694476  123.537968  1.5  6.02  14.07  38.15   
1     11020  447886.0  47.183639  150.920124  1.5  6.05  14.07  38.21   
2     11030  409978.0  36.418986   91.815105  1.5  5.42  12.05  30.99   
3     11040  414966.0  41.879404  107.084049  1.5  5.43  12.19  33.11   
4     11050  386868.0  45.669953  125.916497  1.5  4.80  11.94  32.98   
5     11060  415093.0  40.631885  103.262146  1.5  5.43  12.10  32.85   
6     11070  340286.0  41.653042  123.414893  1.5  4.53   9.84  28.54   
7     11080  408110.0  44.970384  124.877097  1.5  4.78  11.91  32.83   
8     11090  318479.0  39.190099  104.252526  1.5  4.43   9.59  28.78   
9     11100  286648.0  38.268616  100.928222  1.5  4.10   9.50  28.25   
10    11110  374034.0  54.399453  179.338724  1.5  4.67  11.42  34.42   
11    11120  330509.0  51.689797  171.120698  1.5  4.39   9.53  28.60   
12    11130  393060.0  43.150071  107.408371  1.5  4.79  11.96  32.98   
13    11140  435551.0  55.446820  185.163385  1.5  5.99  13.67  38.75   
14    11150  319485.0  47.473436  129.906390  1.5  4.56  10.31  32.74   
15    11160  382584.0  62.125087  204.896804  1.5  4.70  11.49  32.75   
16    11170  353379.0  44.465770  118.264452  1.5  4.71  11.49  32.74   
17    11180  266505.0  32.100454   77.200055  1.5  4.10   9.34  26.19   
18    11190  447725.0  59.129038  188.562514  1.5  6.05  15.17  42.09   
19    11200  389260.0  41.546157  101.797524  1.5  4.76  11.87  32.77   
20    11210  379023.0  50.625584  146.211666  1.5  4.75  11.99  35.36   
21    11220  456132.0  68.428545  202.701807  1.5  6.05  15.16  42.74   
22    11230  504125.0  96.164419  334.103683  1.5  6.58  18.03  53.72   
23    11240  420383.0  85.713394  301.722096  1.5  5.73  13.18  38.99   
24    11250  333575.0  67.258181  226.267210  1.5  4.52   9.81  31.44   

         max  
0    6531.17  
1    8133.58  
2    5238.22  
3    4551.39  
4    2824.50  
5    2949.91  
6    4163.46  
7    4730.88  
8    4548.29  
9    3417.50  
10   5987.41  
11   5660.62  
12   4571.02  
13  12311.75  
14   5007.51  
15   6477.78  
16   2558.08  
17   2496.35  
18   7951.67  
19   3663.11  
20   5480.66  
21   5266.04  
22  11121.51  
23   7071.22  
24   6439.83

In [179]:
display(df_sum.head())
display(df_mean.head())

도착자치구코드    이동시간합계       이동인구합계
0    11010  26504649  19803049.50
1    11020  26065980  21132891.17
2    11030  23776976  14930983.13
3    11040  24144722  17378528.65
4    11050  23556776  17668243.56

도착자치구코드     이동시간평균     이동인구평균
0    11010  59.819645  44.694476
1    11020  58.197800  47.183639
2    11030  57.995736  36.418986
3    11040  58.184820  41.879404
4    11050  60.890991  45.669953

In [212]:
# 파일을 합쳐볼까요
# pd.merge()
merged_data = pd.merge(df_sum, df_mean, on='도착자치구코드')
# df_mean
# df_sum
merged_data.to_csv('./seoul_moving_day.csv')
merged_data.head()

# pd.merge(merged_data,moving,left_on='도착자치구코드', right_on='자치구').drop(columns=['자치구'])
# 도착지역이동인구 -> 지역대중교통필요수요

도착자치구코드    이동시간합계       이동인구합계     이동시간평균     이동인구평균
0    11010  26504649  19803049.50  59.819645  44.694476
1    11020  26065980  21132891.17  58.197800  47.183639
2    11030  23776976  14930983.13  57.995736  36.418986
3    11040  24144722  17378528.65  58.184820  41.879404
4    11050  23556776  17668243.56  60.890991  45.669953

In [ ]:
# 해당 데이터프레임을 csv 파일로 저장하세요
# to_csv()


---

# 3.데이터 분석하기
+ KeyPoint : 데이터의 형태를 살펴보고 다양한 분석기법을 통해 모델링에 적합하도록 정제요소를 선별할 수 있다.
  * 데이터들의 패턴 탐색
  * 변수들간의 관계 파악

In [213]:
# 시각화 한글폰트 설정
import seaborn as sns
import matplotlib.pyplot as plt

plt.rc('font', family='Malgun Gothic')
sns.set(font="Malgun Gothic",#"NanumGothicCoding", 
        rc={"axes.unicode_minus":False}, # 마이너스 부호 깨짐 현상 해결
        style='darkgrid')

#### [실습문제10] 데이터 분포 알아보기
* 다양한 변수를 기준으로 그래프를 그려보고 인사이트를 도출해보세요.

In [ ]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.


In [ ]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.


In [ ]:
# 아래에 실습코드를 작성하고 결과를 확인합니다.


In [ ]:
# 위 차트를 통해 알게된 사실을 정리해봅시다.
# 1. 
# 2.
# 3.

---